src/utils/parser/scopes.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+
+import toPairs from "lodash/toPairs";
+import uniq from "lodash/uniq";
+
+import type { Scope } from "./types";
+
+function getScopeVariables(scope: Scope) {
+  const { bindings } = scope;
+
+  return toPairs(bindings).map(([name, binding]) => ({
+    name,
+    references: binding.referencePaths
+  }));
+}
+
+function getScopeChain(scope: Scope): Scope[] {
+  const scopes = [scope];
+
+  do {
+    scopes.push(scope);
+  } while ((scope = scope.parent));
```

Comment:
also one more question, do we always want to push at least one scope in? or should we not push a scope if 
```
// while ((scope = scope.parent))
let parentScope = scope.parent;
if (parentScope !== undefined) return "do nothing";

// or do we want (assume returnOnce only does it once
if (parentScope !== undefined) returnOnce(() => scopes.push(parentScope);
```